sympy/ntheory/residue_ntheory.py

```diff
@@ -1249,6 +1256,11 @@ class mobius(Function):
 
     """
     @classmethod
+    @deprecated("""\
+    The `sympy.ntheory.residue_ntheory.mobius` has been moved to `sympy.functions.combinatorial.numbers.mobius`.""",
+    deprecated_since_version="1.13",
+    active_deprecations_target='deprecated-ntheory-symbolic-functions',
+    stacklevel=7)
     def eval(cls, n):
```

Comment:
Rather than deprecate the `eval` method I would just replace this class with a function that is deprecated. That way we can remove the potentially circular imports now by using delayed imports inside the function.

Reply:
I see. Is this what you mean?

> We could leave behind a function like:
> 
> ```
> @deprecated
> def mobius(n):
>     """
>     Deprecated: use from sympy import mobius instead.
>     """
>     from sympy import mobius as _mobius
>     return _mobius(n)
> ```